src/codegen/intermediate_representation.c

```diff
@@ -478,6 +485,8 @@ IRInstruction *ir_load
   INSTRUCTION(load, IR_LOAD);
 
   load->operand = address;
+  load->type = address->type;
```

Comment:
IIRC this is due to the fact that when we load a variable, we actually need to load the value of the @integer. So it is kind of odd, but I think this is correct. Because a load of @integer type actually loads a static ref or alloca instruction, and returns the @integer. Maybe I'm incorrect here though (I'm on my phone so can't have all the context on screen at once, I just can't wait any longer to check out the review haha)

Reply:
The ‘address’ of a static ref or alloca is known at compile time. When emitting a load to either, the result should be the value stored at that static ref or alloca. This is fine *at the moment* because the type of a static ref or alloca is not a pointer to the storage but rather a value type. For example:
```
;; Current Implementation: 
%1 = alloca integer   ;; %1 is of type `integer`
%2 = load %1             ;; %2 has the same type as %1; that’s fine.

;; Proposed:
%1 = alloca integer  ;; %1 is of type `@integer`
%2 = load %1            ;; We’re loading from a `@integer`; the result should be an `integer`.
```
This is why it’s fine to copy the type atm, but if we do the latter (which makes more sense imo because the ‘value’
of an `alloca` is the address, not the value stored at that address), then we need to change this

Reply:
> Because a load of @integer type actually loads a static ref or alloca instruction, and returns the @integer
My main point is that this is wrong. The *value* of an `alloca` is the `@integer`. *Loading* from the  `alloca`
yields the value stored at that address, i.e. an `integer`. Because, if loading from an `alloca` returned the address, exactly what would the value of the `alloca` be?

Reply:
That actually clears up a lot of things. So a type of alloca instruction generated from @integer variable would actually be @@integer

Reply:
Exactly. The variable (i.e. memory) itself holds a `@integer`, so a pointer to it would be an `@@integer`.

Reply:
This has been done as of https://github.com/LensPlaysGames/FUNCompiler/commit/9b30942185582a6b56bff0d537b716e7130e6b02

Although I'll be the first to admit it's not pretty